pkg/kubectl/cmd/auth/cani.go

```diff
@@ -393,3 +399,22 @@ func isNamespaced(gvr schema.GroupVersionResource, discoveryClient discovery.Dis
 
 	return false, fmt.Errorf("the server doesn't have a resource type '%s' in group '%s'", gvr.Resource, gvr.Group)
 }
+
+func isVerb(o *CanIOptions) bool {
+	nonResourceURLVerbs := []string{"get", "put", " post", "head", "options", "delete", " patch"}
```

Comment:
* split these to `isKnownNonResourceVerb`
* there are typos with leading spaces in the post and patch strings

Reply:
can construct package variable `nonResourceURLVerbs := sets.NewString(...)` and do `nonResourceURLVerbs.Has(verb)`